### PR TITLE
Fix remaining issues for #9195

### LINKED ISSUE #9418: Fix remaining issues for #9195
#9195 was merged because it blocked a lot of other work, but I still had a whole bunch of comments that I wanted to see addressed: https://github.com/dealii/dealii/pull/9195#pullrequestreview-346188824. 

This issue simply reminds us of this.

@peterrum: FYI.

### PR DESCRIPTION
**Pull request by peterrum**
... closes #9418.
 
ping @bangerth

### CODE REVIEW

File: include/deal.II/grid/tria_description.h [hunk 1]
```diff
@@ -448,27 +448,27 @@ namespace TriangulationDescription
 
 
     /**
-     * Construct a construction::Description. In contrast
+     * Construct a TriangulationDescription::Description. In contrast
      * to the function above, this function is also responsible for creating
      * a serial triangulation and for its partitioning (by calling the
-     * provided std::functions). Internally only selected processes (every
-     * n-th/each root of a group of size group_size) create a serial
+     * provided `std::functions' objects). Internally only selected processes (
```
Comment (kronbichler) [inline include/deal.II/grid/tria_description.h]:
```suggestion
     * provided `std::function` objects). Internally only selected processes
```
?
And move the final brace to the next line.

Comment (peterrum) [reply include/deal.II/grid/tria_description.h]: @kronbichler Done.

File: include/deal.II/grid/tria_description.h [hunk 2]
```diff
@@ -271,16 +271,16 @@ namespace TriangulationDescription
    * the cell id, the subdomain_id and the level_subdomain_id as well as
    * information related to manifold_id and boundary_id.
    *
+   * @note This struct stores similarly to ::CellData information regard a cell,
+   *   however, also stores a unique id, partitioning information, and
+   *   information related to the cell faces and edges normally stored
+   *   in ::SubCellData.
```
Comment (bangerth) [inline include/deal.II/grid/tria_description.h]: This reads awkward. How about this:

```
Similarly to dealii::CellData, this structure stores information about a cell. However,
in contrast to dealii::CellData, it also stores a unique id, partitioning information, and
information related to cell faces and edges.

Comment (peterrum) [reply include/deal.II/grid/tria_description.h]: Thanks! Done

**Pull request merged**